warp10/src/main/java/io/warp10/script/functions/GUARD.java

```diff
@@ -0,0 +1,108 @@
+//
+//   Copyright 2021  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script.functions;
+
+import java.util.List;
+
+import io.warp10.script.NamedWarpScriptFunction;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.WarpScriptStack.Macro;
+import io.warp10.script.WarpScriptStackFunction;
+import io.warp10.script.WarpScriptStopException;
+
+public class GUARD extends NamedWarpScriptFunction implements WarpScriptStackFunction {
+
+  public GUARD(String name) {
+    super(name);
+  }
+
+  @Override
+  public Object apply(WarpScriptStack stack) throws WarpScriptException {
+
+    Object top = stack.pop();
+    Object hide = null;
+    boolean hasHide = false;
+
+    if (null == top || top instanceof Long) {
+      hide = top;
+      hasHide = true;
+      top = stack.pop();
+    }
+
+    List<Object> symbols = null;
+
+    if (top instanceof List) {
+      symbols = (List<Object>) top;
+      top = stack.pop();
+    }
```

Comment:
We also could give a context to do the following:
```
<%
  SAVE '.context' STORE
  <%
    // The code
  %>
  $.context GUARD
%>
```

This would be almost equivalent to a RESTORE in a finally clause the difference being we are sure that the context is restored. In the case of a finally clause, this execution could be stopped if the stack is signaled.